### PR TITLE
feat: Allow skipping setup

### PR DESCRIPTION
**Pull request by rvesse**
If a user is invoking the action multiple times then the trivy binary gets installed multiple times.  Users can avoid this by managing the installation themselves and setting the skip-setup input to true, or by letting the action install in on their first invocation and then setting skip-setup to true on subsequent invocations

This is an alternative solution to aquasecurity/setup-trivy#7 as discussed in that PR to address aquasecurity/setup-trivy#6

By allowing users to control whether the call to the `setup-trivy` action is made they can control how the `trivy` binary is installed, and avoid installing it multiple times if they have workflows that call `trivy` multiple times.

Added an example test at https://github.com/rvesse/setup-trivy-debugging/blob/main/.github/workflows/setup-trivy-indirect-fixed.yml with job output at https://github.com/rvesse/setup-trivy-debugging/actions/runs/11342576655/job/31543286568 where we can see that Trivy is no longer installed multiple times

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -681,6 +681,7 @@ Following inputs can be used as `step.with` keys:
 | `limit-severities-for-sarif` | Boolean | false                              | By default *SARIF* format enforces output of all vulnerabilities regardless of configured severities. To override this behavior set this parameter to **true** |
 | `docker-host`                | String  |                                    | By default it is set to `unix://var/run/docker.sock`, but can be updated to help with containerized infrastructure values                                      |
 | `version`                    | String  | `v0.56.1`                          | Trivy version to use, e.g. `latest` or `v0.56.1`                                                                                                               |
+| `skip-setup`                 | Boolean | false                              | Whether to skip installing the `trivy` binary, if you are installing it via your own means, or are calling this action multiple times in a workflow then you may wish to set this to true on subsequent calls. |
```
Comment (DmitriyLewen) [inline README.md:684]: `skip-setup` is not entirely clear outside the context of these PRs.
What if we use `skip-setup-trivy`?

Comment (DmitriyLewen) [reply README.md:684]: I think `skip running ` trivy-setup` to install `Trivy` is enough.

**Review by DmitriyLewen — COMMENTED** [review]
Can you add examples to `README.md` using `setup-trivy` action + `skip-setup-trivy` input?

**Comment by rvesse** [comment]
> Can you add examples to `README.md` using `setup-trivy` action + `skip-setup-trivy` input?

Yes, added examples of both calling `setup-trivy` directly and relying on the implicit call to it

**Review by DmitriyLewen — APPROVED** [review]
LGTM

@simar7 wdyt?

**Review by simar7 — APPROVED** [review]
lgtm!

**Comment by antoninbas** [comment]
Would love to use this. Could we have a v0.29.0 release for the action?

**Comment by DmitriyLewen** [comment]
Hello @antoninbas
Unfortunately, we are busy with other tasks.

But you can use commit number or master branch.

**Pull request merged**